src/test/java/org/folio/innreach/domain/listener/KafkaCirculationEventListenerApiTest.java

```diff
@@ -314,6 +316,26 @@ void shouldUpdateTransactionWhenCancelRequest() {
     assertEquals(CANCEL_REQUEST, updatedTransaction.getState());
   }
 
+  @Test
+  @Sql(scripts = {
+    "classpath:db/central-server/pre-populate-central-server.sql",
+    "classpath:db/inn-reach-transaction/pre-populate-another-inn-reach-transaction.sql"
+  })
+  void shouldUpdateTransactionWithPatronHoldWhenCancelRequest() {
+    var event = getRequestDomainEvent(DomainEventType.UPDATED);
+    var request = event.getData().getNewEntity();
+    request.setId(PRE_POPULATED_PATRON_TRANSACTION_REQUEST_ID);
+    request.setStatus(CLOSED_CANCELLED);
+
+    listener.handleRequestEvents(asSingleConsumerRecord(CIRC_REQUEST_TOPIC, REQUEST_ID, event));
```

Comment:
use same id for event key as it's a bit confusing why they are different

Reply:
Fixed